db/db_utils.go

```diff
@@ -252,19 +253,17 @@ func UpdateContext(ctx context.Context, collection string, query any, update any
 }
 
 // ReplaceContext replaces one matching document in the collection.
-func ReplaceContext(ctx context.Context, collection string, query any, replacement any) error {
+func ReplaceContext(ctx context.Context, collection string, query any, replacement any) (*db.ChangeInfo, error) {
 	res, err := evergreen.GetEnvironment().DB().Collection(collection).ReplaceOne(ctx,
 		query,
 		replacement,
+		options.Replace().SetUpsert(true),
```

Comment:
Should we make a comment for why we need this to always upsert, or consider just passing this as an options parameter?

Otherwise, I'd be fine with just adding a note (or maybe call it `ReplaceWithUpsertContext`) that callers shouldn't use this if they don't want to upsert when the doc doesn't exist.

Reply:
I'll add a comment, before this PR we used the anser's Upsert which under the hood (if the update was a raw document rather than a proper query e.g. "$set"), called Replace with SetUpsert to true.

With this PR, I've migrated all usages like that to just call this ReplaceContext (which was only used by tests before).